File: Domain/Models/Vehiculos/Modelo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Domain.Models.Vehiculos
{
    public class Modelo
    {

        [Key]
        public int ModeloId { get; set; }

        [Required]
        [Column(TypeName = "nvarchar(200)")]
        public string Descripcion { get; set; }
    }
}

File: CarAgency/ClientApp/src/domain/models/vehiculos/Modelo.ts
import { IModelo } from "domain/interfaces/vehiculos/IModelo";
import {TipoVehiculo } from "domain/models/vehiculos/TipoVehiculo";

export class Modelo implements IModelo {
    constructor() {
        this.ModeloId = 0;
        this.Descripcion = "";
        this.TipoVehiculo = new TipoVehiculo();
    }
    TipoVehiculo: TipoVehiculo;
    ModeloId?: number;
    Descripcion : string;
}